src/transformStateWithClones.js

```diff
@@ -8,6 +8,36 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  let stateClone = { ...state };
+  const array = [];
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'clear':
+        stateClone = {};
+        break;
+
+      case 'addProperties':
+        stateClone = { ...stateClone, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        stateClone = { ...stateClone };
+
+        action.keysToRemove.forEach((key) => {
+          delete stateClone[key];
+        });
+
+        break;
+
+      default:
+        throw new Error('Error');
+    }
```

Comment:
While you have correctly used a switch statement to handle different action types, it is recommended to place the code that copies the state (`array.push({ ...stateClone })`) outside of the switch block, but still inside the loop, to adhere to the DRY principle. This way, you won't have to repeat the copying operation in each case.